### ISSUE TITLE
[PATCH] Upgrade to threads::shared 1.53

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#15784]
Migrated from [rt.perl.org#130457](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=130457) (status was 'resolved')

Searchable as RT130457$

**Comment by p5pRT** [comment]
### From @jdhedden

This is a bug report for perl from jdhedden@&#8203;cpan\.org\,
generated with the help of perlbug 1\.40 running under perl 5\.24\.1\.

The attached patch upgrades 'threads&#8203;::shared' to version 1\.53\.

This is a sync with blead in preparation for a CPAN release\.
No functional changes\.


---
Flags&#8203;:
&nbsp;    category=library
&nbsp;    severity=low
&nbsp;    Type=Patch
&nbsp;    PatchStatus=HasPatch
&nbsp;    module=threads&#8203;::shared


---
Site configuration information for perl 5\.24\.1&#8203;:

Configured by Debian Project at Sat Nov 26 21&#8203;:12&#8203;:44 UTC 2016\.

Summary of my perl5 \(revision 5 version 24 subversion 1\) configuration&#8203;:
&nbsp;   
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=3\.16\.0\, archname=x86\_64\-linux\-gnu\-thread\-multi
&nbsp;    uname='linux localhost 3\.16\.0 \#1 smp debian 3\.16\.0 x86\_64 gnulinux '
&nbsp;    config\_args='\-Dusethreads \-Duselargefiles \-Dcc=x86\_64\-linux\-gnu\-gcc \-Dcpp=x86\_64\-linux\-gnu\-cpp \-Dld=x86\_64\-linux\-gnu\-gcc \-Dccflags=\-DDEBIAN \-Wdate\-time \-D\_FORTIFY\_SOURCE=2 \-g \-O2 \-fdebug\-prefix\-map=/build/perl\-BIBcGN/perl\-5\.24\.1~rc4=\. \-fstack\-protector\-strong \-Wformat \-Werror=format\-security \-Dldflags= \-Wl\,\-z\,relro \-Dlddlflags=\-shared \-Wl\,\-z\,relro \-Dcccdlflags=\-fPIC \-Darchname=x86\_64\-linux\-gnu \-Dprefix=/usr \-Dprivlib=/usr/share/perl/5\.24 \-Darchlib=/usr/lib/x86\_64\-linux\-gnu/perl/5\.24 \-Dvendorprefix=/usr \-Dvendorlib=/usr/share/perl5 \-Dvendorarch=/usr/lib/x86\_64\-linux\-gnu/perl5/5\.24 \-Dsiteprefix=/usr/local \-Dsitelib=/usr/local/share/perl/5\.24\.1 \-Dsitearch=/usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1 \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/share/man/man3 \-Dsiteman1dir=/usr/local/man/man1 \-Dsiteman3dir=/usr/local/man/man3 \-Dusesitecustomize \-Duse64bitint \-Dman1ext=1 \-Dman3ext=3perl
\-Dpager=/usr/bin/sensible\-pager \-Uafs \-Ud\_csh \-Ud\_ualarm \-Uusesfio \-Uusenm \-Ui\_libutil \-Uversiononly \-DDEBUGGING=\-g \-Doptimize=\-O2 \-dEs \-Duseshrplib \-Dlibperl=libperl\.so\.5\.24\.1'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=define\, usemultiplicity=define
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='x86\_64\-linux\-gnu\-gcc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fwrapv \-fno\-strict\-aliasing \-pipe \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
&nbsp;    optimize='\-O2 \-g'\,
&nbsp;    cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DDEBIAN \-fwrapv \-fno\-strict\-aliasing \-pipe \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='6\.2\.1 20161119'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678\, doublekind=3
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16\, longdblkind=3
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='x86\_64\-linux\-gnu\-gcc'\, ldflags =' \-fstack\-protector\-strong \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /usr/lib/gcc/x86\_64\-linux\-gnu/6/include\-fixed /usr/include/x86\_64\-linux\-gnu /usr/lib /lib/x86\_64\-linux\-gnu /lib/\.\./lib /usr/lib/x86\_64\-linux\-gnu /usr/lib/\.\./lib /lib
&nbsp;    libs=\-lgdbm \-lgdbm\_compat \-ldb \-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    perllibs=\-ldl \-lm \-lpthread \-lc \-lcrypt
&nbsp;    libc=libc\-2\.24\.so\, so=so\, useshrplib=true\, libperl=libperl\.so\.5\.24
&nbsp;    gnulibc\_version='2\.24'
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib \-fstack\-protector\-strong'

Locally applied patches&#8203;:
&nbsp;    RC4
&nbsp;    DEBPKG&#8203;:debian/cpan\_definstalldirs \- Provide a sensible INSTALLDIRS default for modules installed from CPAN\.
&nbsp;    DEBPKG&#8203;:debian/db\_file\_ver \- http&#8203;://bugs\.debian\.org/340047 Remove overly restrictive DB\_File version check\.
&nbsp;    DEBPKG&#8203;:debian/doc\_info \- Replace generic man\(1\) instructions with Debian\-specific information\.
&nbsp;    DEBPKG&#8203;:debian/enc2xs\_inc \- http&#8203;://bugs\.debian\.org/290336 Tweak enc2xs to follow symlinks and ignore missing @&#8203;INC directories\.
&nbsp;    DEBPKG&#8203;:debian/errno\_ver \- http&#8203;://bugs\.debian\.org/343351 Remove Errno version check due to upgrade problems with long\-running processes\.
&nbsp;    DEBPKG&#8203;:debian/libperl\_embed\_doc \- http&#8203;://bugs\.debian\.org/186778 Note that libperl\-dev package is required for embedded linking
&nbsp;    DEBPKG&#8203;:fixes/respect\_umask \- Respect umask during installation
&nbsp;    DEBPKG&#8203;:debian/writable\_site\_dirs \- Set umask approproately for site install directories
&nbsp;    DEBPKG&#8203;:debian/extutils\_set\_libperl\_path \- EU&#8203;:MM&#8203;: set location of libperl\.a under /usr/lib
&nbsp;    DEBPKG&#8203;:debian/no\_packlist\_perllocal \- Don't install \.packlist or perllocal\.pod for perl or vendor
&nbsp;    DEBPKG&#8203;:debian/fakeroot \- Postpone LD\_LIBRARY\_PATH evaluation to the binary targets\.
&nbsp;    DEBPKG&#8203;:debian/instmodsh\_doc \- Debian policy doesn't install \.packlist files for core or vendor\.
&nbsp;    DEBPKG&#8203;:debian/ld\_run\_path \- Remove standard libs from LD\_RUN\_PATH as per Debian policy\.
&nbsp;    DEBPKG&#8203;:debian/libnet\_config\_path \- Set location of libnet\.cfg to /etc/perl/Net as /usr may not be writable\.
&nbsp;    DEBPKG&#8203;:debian/mod\_paths \- Tweak @&#8203;INC ordering for Debian
&nbsp;    DEBPKG&#8203;:debian/prune\_libs \- http&#8203;://bugs\.debian\.org/128355 Prune the list of libraries wanted to what we actually need\.
&nbsp;    DEBPKG&#8203;:fixes/net\_smtp\_docs \- \[rt\.cpan\.org \#36038\] http&#8203;://bugs\.debian\.org/100195 Document the Net&#8203;::SMTP 'Port' option
&nbsp;    DEBPKG&#8203;:debian/perlivp \- http&#8203;://bugs\.debian\.org/510895 Make perlivp skip include directories in /usr/local
&nbsp;    DEBPKG&#8203;:debian/deprecate\-with\-apt \- http&#8203;://bugs\.debian\.org/747628 Point users to Debian packages of deprecated core modules
&nbsp;    DEBPKG&#8203;:debian/squelch\-locale\-warnings \- http&#8203;://bugs\.debian\.org/508764 Squelch locale warnings in Debian package maintainer scripts
&nbsp;    DEBPKG&#8203;:debian/skip\-upstream\-git\-tests \- Skip tests specific to the upstream Git repository
&nbsp;    DEBPKG&#8203;:debian/patchlevel \- http&#8203;://bugs\.debian\.org/567489 List packaged patches for 5\.24\.1~rc4\-1 in patchlevel\.h
&nbsp;    DEBPKG&#8203;:debian/skip\-kfreebsd\-crash \- http&#8203;://bugs\.debian\.org/628493 \[perl \#96272\] Skip a crashing test case in t/op/threads\.t on GNU/kFreeBSD
&nbsp;    DEBPKG&#8203;:fixes/document\_makemaker\_ccflags \- http&#8203;://bugs\.debian\.org/628522 \[rt\.cpan\.org \#68613\] Document that CCFLAGS should include $Config\{ccflags\}
&nbsp;    DEBPKG&#8203;:debian/find\_html2text \- http&#8203;://bugs\.debian\.org/640479 Configure CPAN&#8203;::Distribution with correct name of html2text
&nbsp;    DEBPKG&#8203;:debian/perl5db\-x\-terminal\-emulator\.patch \- http&#8203;://bugs\.debian\.org/668490 Invoke x\-terminal\-emulator rather than xterm in perl5db\.pl
&nbsp;    DEBPKG&#8203;:debian/cpan\-missing\-site\-dirs \- http&#8203;://bugs\.debian\.org/688842 Fix CPAN&#8203;::FirstTime defaults with nonexisting site dirs if a parent is writable
&nbsp;    DEBPKG&#8203;:fixes/memoize\_storable\_nstore \- \[rt\.cpan\.org \#77790\] http&#8203;://bugs\.debian\.org/587650 Memoize&#8203;::Storable&#8203;: respect 'nstore' option not respected
&nbsp;    DEBPKG&#8203;:debian/regen\-skip \- Skip a regeneration check in unrelated git repositories
&nbsp;    DEBPKG&#8203;:debian/makemaker\-pasthru \- http&#8203;://bugs\.debian\.org/758471 Pass LD settings through to subdirectories
&nbsp;    DEBPKG&#8203;:debian/makemaker\-manext \- http&#8203;://bugs\.debian\.org/247370 Make EU&#8203;::MakeMaker honour MANnEXT settings in generated manpage headers
&nbsp;    DEBPKG&#8203;:debian/devel\-ppport\-reproducibility \- http&#8203;://bugs\.debian\.org/801523 Sort the list of XS code files when generating RealPPPort\.xs
&nbsp;    DEBPKG&#8203;:debian/encode\-unicode\-bom\-doc \- http&#8203;://bugs\.debian\.org/798727 Document Debian backport of Encode&#8203;::Unicode fix
&nbsp;    DEBPKG&#8203;:debian/kfreebsd\-softupdates \- http&#8203;://bugs\.debian\.org/796798 Work around Debian Bug\#796798
&nbsp;    DEBPKG&#8203;:fixes/autodie\-scope \- http&#8203;://bugs\.debian\.org/798096 Fix a scoping issue with "no autodie" and the "system" sub
&nbsp;    DEBPKG&#8203;:fixes/crosscompile\-no\-targethost \- \[23695c0\] \[perl \#127234\] Fix the Configure escape with usecrosscompile but no targethost
&nbsp;    DEBPKG&#8203;:fixes/memoize\-pod \- \[rt\.cpan\.org \#89441\] Fix POD errors in Memoize
&nbsp;    DEBPKG&#8203;:fixes/ok\-pod \- Added encoding for pod\.
&nbsp;    DEBPKG&#8203;:debian/hurd\-softupdates \- http&#8203;://bugs\.debian\.org/822735 Fix t/op/stat\.t failures on hurd
&nbsp;    DEBPKG&#8203;:fixes/nntp\_docs \- http&#8203;://bugs\.debian\.org/51962 Net&#8203;::NNTP&#8203;: Correct innd/nnrpd confusion in relation to Reader option
&nbsp;    DEBPKG&#8203;:fixes/math\_complex\_doc\_great\_circle \- http&#8203;://bugs\.debian\.org/697567 \[rt\.cpan\.org \#114104\] Math&#8203;::Trig&#8203;: clarify definition of great\_circle\_midpoint
&nbsp;    DEBPKG&#8203;:fixes/math\_complex\_doc\_see\_also \- http&#8203;://bugs\.debian\.org/697568 \[rt\.cpan\.org \#114105\] Math&#8203;::Trig&#8203;: add missing SEE ALSO
&nbsp;    DEBPKG&#8203;:fixes/math\_complex\_doc\_angle\_units \- http&#8203;://bugs\.debian\.org/731505 \[rt\.cpan\.org \#114106\] Math&#8203;::Trig&#8203;: document angle units
&nbsp;    DEBPKG&#8203;:fixes/cpan\_web\_link \- http&#8203;://bugs\.debian\.org/367291 CPAN&#8203;: Add link to main CPAN web site
&nbsp;    DEBPKG&#8203;:fixes/time\_piece\_doc \- http&#8203;://bugs\.debian\.org/817925 Time&#8203;::Piece&#8203;: Improve documentation for add\_months and add\_years
&nbsp;    DEBPKG&#8203;:fixes/perlbug\-refactor \- http&#8203;://bugs\.debian\.org/822463 \[perl \#128020\] perlbug&#8203;: Refactor duplicated file reading code
&nbsp;    DEBPKG&#8203;:fixes/perlbug\-linewrap \- http&#8203;://bugs\.debian\.org/822463 \[perl \#128020\] perlbug&#8203;: wrap overly long lines
&nbsp;    DEBPKG&#8203;:fixes/hurd\_sigaction \- http&#8203;://bugs\.debian\.org/825016 \[d54f4ed\] ext/POSIX/t/sigaction\.t&#8203;: Skip uid and pid tests on GNU/Hurd
&nbsp;    DEBPKG&#8203;:fixes/hurd\_hints \- \[4694301\] http&#8203;://bugs\.debian\.org/825020 \[perl \#128279\] Modify hints for Hurd per Debian ticket 825020\.
&nbsp;    DEBPKG&#8203;:fixes/extutils\-parsexs\-reproducibility \- \[perl \#128517\] http&#8203;://bugs\.debian\.org/829296 Make the output of ExtUtils&#8203;::ParseXS reproducible
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/sitecustomize\-in\-etc \- Look for sitecustomize\.pl in /etc/perl rather than sitelib on Debian systems
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/test\-suite\-without\-dot \- \[perl \#127810\] Patch unit tests to explicitly insert "\." into @&#8203;INC when needed\.
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/eumm\-without\-dot \- \[perl \#127810\] Add PERL\_USE\_UNSAFE\_INC support to EU&#8203;::MM for fortify\_inc support\.
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/cpan\-without\-dot \- \[perl \#127810\] Set PERL\_USE\_UNSAFE\_INC for cpan usage
&nbsp;    DEBPKG&#8203;:debian/document\_inc\_removal \- Document in perlvar that we remove '\.' from @&#8203;INC by default
&nbsp;    DEBPKG&#8203;:fixes/extutils\_makemaker\_reproducible \- http&#8203;://bugs\.debian\.org/835815 http&#8203;://bugs\.debian\.org/834190 Make perllocal\.pod files reproducible
&nbsp;    DEBPKG&#8203;:debian/CVE\-2016\-1238/remove\-inc\-test \- Remove test for '\.' in @&#8203;INC as it might not be
&nbsp;    DEBPKG&#8203;:debian/customized \- Update customized\.dat for files patched in Debian
&nbsp;    DEBPKG&#8203;:fixes/file\_path\_hurd\_errno \- File\-Path&#8203;: Fix test failure in Hurd due to hard\-coded ENOENT
&nbsp;    DEBPKG&#8203;:debian/hppa\_op\_optimize\_workaround \- http&#8203;://bugs\.debian\.org/838613 Temporarily lower the optimization of op\.c on hppa due to gcc\-6 problems
&nbsp;    DEBPKG&#8203;:fixes/test\-builder\-warning \- http&#8203;://bugs\.debian\.org/840968 Silence a 'used only once' warning in Test&#8203;::Builder
&nbsp;    DEBPKG&#8203;:fixes/longdblinf\-randomness \- \[perl \#130133\] http&#8203;://bugs\.debian\.org/844752 Configure&#8203;: fix garbage filtering with 80\-bit long doubles
&nbsp;    DEBPKG&#8203;:debian/installman\-utf8 \- http&#8203;://bugs\.debian\.org/840211 Generate man pages with UTF\-8 characters



---
@&#8203;INC for perl 5\.24\.1&#8203;:
&nbsp;    /etc/perl
&nbsp;    /usr/local/lib/x86\_64\-linux\-gnu/perl/5\.24\.1
&nbsp;    /usr/local/share/perl/5\.24\.1
&nbsp;    /usr/lib/x86\_64\-linux\-gnu/perl5/5\.24
&nbsp;    /usr/share/perl5
&nbsp;    /usr/lib/x86\_64\-linux\-gnu/perl/5\.24
&nbsp;    /usr/share/perl/5\.24
&nbsp;    /usr/local/lib/site\_perl
&nbsp;    /usr/lib/x86\_64\-linux\-gnu/perl\-base



---
Environment for perl 5\.24\.1&#8203;:
&nbsp;    HOME=/home/jdhedden
&nbsp;    LANG=C
&nbsp;    LANGUAGE=C
&nbsp;    LC\_ALL=C
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/home/jdhedden/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/games
&nbsp;    PERLIO=perlio
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @jdhedden

<details><summary>0001-Upgrade-to-threads-shared-1.53.patch</summary>

```diff
From 3ff1f2b0f3c34a03ec7a671bf161d3f57a2d0c38 Mon Sep 17 00:00:00 2001
From: jdhedden <jdhedden@cpan.org>
Date: Fri, 30 Dec 2016 12:37:21 -0500
Subject: [PATCH] Upgrade to threads::shared 1.53

---
 Porting/Maintainers.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Porting/Maintainers.pl b/Porting/Maintainers.pl
index 723928e9dc..007e8d0570 100755
--- a/Porting/Maintainers.pl
+++ b/Porting/Maintainers.pl
@@ -1257,7 +1257,7 @@ use File::Glob qw(:case);
     },
 
     'threads::shared' => {
-        'DISTRIBUTION' => 'JDHEDDEN/threads-shared-1.52.tar.gz',
+        'DISTRIBUTION' => 'JDHEDDEN/threads-shared-1.53.tar.gz',
         'FILES'        => q[dist/threads-shared],
         'EXCLUDED'     => [
             qw( examples/class.pl
-- 
2.11.0


```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Fri\, 30 Dec 2016 17&#8203;:41&#8203;:16 GMT\, jdhedden@&#8203;cpan\.org wrote&#8203;:
> This is a bug report for perl from jdhedden@&#8203;cpan\.org\,
> generated with the help of perlbug 1\.40 running under perl 5\.24\.1\.
> 
> The attached patch upgrades 'threads&#8203;::shared' to version 1\.53\.
> 
> This is a sync with blead in preparation for a CPAN release\.
> No functional changes\.

Thanks\, applied to blead in commit 8df022406424cfae8a958737434c9949fd7d1bdc\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'
